api/turing/batch/ensembling/runner_test.go

```diff
@@ -84,7 +86,7 @@ func TestRun(t *testing.T) {
 					},
 				}, nil)
 
-				svc.On("FindByID", mock.Anything, mock.Anything).Return(
+				svc.On("FindByID", mock.Anything, mock.Anything, mock.Anything).Return(
```

Comment:
Did you forget to update this mocks after reverting the signature of methods on `EnsemblingJobService`?

Reply:
hmm why didn't the test flag this out, let me check

Reply:
This is not very intuitive but I think I have found out the reason why it does not give a runtime error when more arguments are supplied (not less).

Seems to have an [issue raised with the library here](https://github.com/stretchr/testify/issues/587) and it seems like it is a gotcha with the library when `mock.Anything` is used. The arity of the function is not considered when using svc.On(funcName, mock.Anything).